build.gradle

```diff
@@ -84,21 +86,6 @@ task smoke(type: Test) {
   classpath = sourceSets.smokeTest.runtimeClasspath
 }
 
-checkstyle {
-  maxWarnings = 0
-  toolVersion = '8.29'
-  getConfigDirectory().set(new File(rootDir, 'config/checkstyle'))
-}
-
-pmd {
```

Comment:
do we want to keep PMD enabled? sonarqube is supposed to cover it
moving this to the plugin makes it harder to disable or alter the config in a team project I assume?

Reply:
The default PMD settings can be extended with additional rulesets or replaced completely with custom rulesets if desired in buildscript.

Sonarqube runs PMD if I understand correctly? So we'd still want control of the PMD rulesets. Also does sonarqube run analysis locally or just on CI?

Reply:
no they've written their own static analysis tool afaik, they used to.

It runs on CI but you can run it locally with sonarlint

Reply:
Re: disabling rulesets. If teams have a need to blanket remove an entire rule then it probably shouldn't be in the default HMCTS rulesets (teams can suppress individual violations where necessary).

For example, I did remove one [PMD rule](https://stackoverflow.com/questions/21592497/java-for-each-loop-being-flagged-as-ur-anomaly-by-pmd) that doesn't work correctly with modern Java constructs like foreach loops.

Reply:
There are a lot of PMD rules which report on "issues" we don't really care. I think by default it should be disabled and have an option to enable and override if necessary

Reply:
or just not include it?
@satyachundur ?

Reply:
Agreed we don't want to impose a tool with a bad signal to noise ratio. I'll go through the PMD warnings for a couple of our larger projects and see if it flags up anything useful, and either trim back the ruleset or remove the tool completely.

Reply:
@timja @satyachundur PMD is no longer applied in the java plugin. Are you happy to proceed with this PR as is?